content/academy/pro_apify_scraper_development/integrating_webhooks.md

```diff
@@ -0,0 +1,184 @@
+
+```JavaScript
+const filtered = items.reduce((acc, curr) => {
+    // Grab the price of the item matching our current
+    // item's ASIN in the map. If it doesn't exist, set
+    // "prevPrice" to null
+    const prevPrive = acc?.[curr.asin] ? +acc[curr.asin].offer.slice(1) : null;
+
+    // Grab the price of our current offer
+    const price = +curr.offer.slice(1);
+
+    // If the item doesn't yet exist in the map, add it.
+    // Or, if the current offer's price is less than the
+    // saved one, replace the saved one
+    if (!acc[curr.asin] || prevPrive > price) acc[curr.asin] = curr;
+
+    // Return the map
+    return acc;
+}, {});
+```
+
+The results should be an array, so finally, we can take the map we just created and push an array of all of its values to the actor's default dataset:
+
+```JavaScript
+await Apify.pushData(Object.values(filtered));
+```
+
+Our final code looks like this:
+
+```JavaScript
+const Apify = require('apify');
+
+Apify.main(async () => {
+    const { datasetId } = await Apify.getInput();
+    const dataset = await Apify.openDataset(datasetId);
+
+    const { items } = await dataset.getData();
+
+    const filtered = items.reduce((acc, curr) => {
+        const prevPrive = acc?.[curr.asin] ? +acc[curr.asin].offer.slice(1) : null;
+        const price = +curr.offer.slice(1);
+
+        if (!acc[curr.asin] || prevPrive > price) acc[curr.asin] = curr;
+
+        return acc;
+    }, {});
+
+    await Apify.pushData(Object.values(filtered));
+});
+```
+
+Cool! But **wait**, don't forget to configure the **INPUT_SCHEMA.json** file as well! It's not necessary to do this step, as we'll be calling the actor through Apify's API within a webhook, but it's still good to get into the habit of writing quality input schemas that describe the input values your actors are expecting.
+
+```JSON
+{
+    "title": "Amazon Filter Actor",
+    "type": "object",
+    "schemaVersion": 1,
+    "properties": {
+        "datasetId": {
+            "title": "Dataset ID",
+            "type": "string",
+            "description": "Enter the ID of the dataset.",
+            "editor": "textfield"
+        }
+    },
+    "required": ["datasetId"]
+}
+```
+
+Now we're done, and we can push it up to the Apify platform with the `apify push` command.
+
+## [](#setting-up-the-webhook) Setting up the webhook
+
+Since we'll be calling the actor via the [Apify API](https://docs.apify.com/tutorials/integrations/run-actor-and-retrieve-data-via-api#run-an-actor-or-task), we'll need to grab hold of the ID of the actor we just created and pushed to the platform. The ID is always accessible through the **Settings** page of the actor.
+
+![Actor ID in actor settings]({{@asset pro_apify_scraper_development/images/actor-settings.webp}})
```

Comment:
Could you re-take this screenshot with a wider window, please? The description box in a new line isn't optimal. In fact, you can probably just trim the bottom part of the image off if it isn't essential :) 

<img width="676" alt="Screenshot 2022-04-13 at 10 45 07" src="https://user-images.githubusercontent.com/38246758/163137278-0f5d9838-34bf-44c1-b303-fd5173381176.png">